Feature/view presentations

natyork opened the pull request:
This PR creates a filter for the presentations index view, IOT filter page by presenters , per pivotal story: https://www.pivotaltracker.com/story/show/140356273

jacquesporveau commented:
jenkins test this please

jacquesporveau commented on app/controllers/presentations_controller.rb:
```diff
@@ -1,8 +1,16 @@
-    @presentations = Presentation.joins(:event)
-                                 .merge(Event.order(date: :desc))
-                                 .page(params[:page])
+    if params[:presentation] && params[:presentation][:person_id].present?
```
Brendan has asked me to move to the safe navigation operator. 

http://mitrev.net/ruby/2015/11/13/the-operator-in-ruby/

brendandeere replied:
I would also take a look at the methods ruby hash probides you for doing this.

#fetch and #dig

jacquesporveau commented on app/views/presentations/_filter.html.erb:
```diff
@@ -0,0 +1,7 @@
+<%= form_tag presentations_path, method: 'get' do %>
+  <p>
+  <%= label_tag('Filter by Presenter:') %>
```
Indent.

natyork replied:
I think i need to look at my vim settings, because my indents often un-indent automatically if I make changes to the line for erb files....it must be interpreting the tags as being at the same level? I'll investigate, because you've caught this a couple times.

jacquesporveau commented on app/views/presentations/index.html.erb:
```diff
@@ -23,6 +23,9 @@
   </tbody>
   <tfoot>
     <tr>
+      <td><%= render 'filter' %></td>
```
What about the idea of having this filter at the top somewhere. The table will differ in row count making the filter show up at different places down the page. If it is above all the rows it will always been in the same place.

natyork replied:
good point, i like that idea

graygilmore commented on app/views/presentations/_filter.html.erb:
```diff
@@ -0,0 +1,7 @@
+<%= form_tag presentations_path, method: 'get' do %>
+  <p>
+    <%= label_tag('Filter by Presenter:') %>
+    <%= collection_select(:presentation, :person_id, Person.all, :id, :first_name, prompt: true) %>
```
Does this allow us to also set a "All Presenters" option?

natyork replied:
I've added in `include_blank: 'All Presenters' which allows us to filter for All presenters in the controller.

natyork commented:
Jenkins failing due to Rubocop Ruby parser defaulting to v 2.1. PR #55 updates the .rubocop.yml file to set TargetRubyVersion to 2.3.1. When #55 is merged, there should be no rubocop offenses for the safe navigation operator.

qr8r commented:
@natyork Since PR #55 is blocking future PRS, can someone extract the changes required to rubycop into their own pr that I can merge separately from the new features being added there?

qr8r commented:
This is great! It definitely works. I mentioned the other day there is a fairly common rails tool for implementing this. I was wondering if you had been able to find it.

natyork commented:
@brendandeere I did look at Ransack as an option, but I thought it might be more than we needed for this app. However, if you think that is a more sensible/flexible approach, I could use that instead!

qr8r commented:
Yeah, I think I would like to see Ransack implemented. Or at the very least if its not something we wanted to do now,  I would like to see the ActiveRecord methods moved out of the controller and into a scope on Presentation.

natyork commented:
@brendandeere, this PR has been updated to make use of Ransack for the filter feature.

jacquesporveau commented on app/controllers/presentations_controller.rb:
```diff
@@ -1,8 +1,10 @@
-    @presentations = Presentation.joins(:event)
-                                 .merge(Event.order(date: :desc))
-                                 .page(params[:page])
+    @q = Presentation.ransack(params[:q])
```
I'd like this instance variable named a little bit more intuitively. I am guessing it means query but I don't know if that's what it is.

natyork replied:
It's a ransack default search parameter. You can customize it, but I left the default for the time being to minimize query string length.
https://github.com/activerecord-hackery/ransack/tree/v1.8.2#simple-mode

natyork commented:
Split into two commits and updated the search key to something more readable per @jacquesporveau suggestions . Waiting on @brendandeere review before merging.
